notebook/notebookapp.py

```diff
@@ -2289,14 +2319,16 @@ def list_running_servers(runtime_dir=None):
     if not os.path.isdir(runtime_dir):
         return
 
+    api_version_json_bytes = json.dumps({"version":__version__}).encode()
     for file_name in os.listdir(runtime_dir):
         if re.match('nbserver-(.+).json', file_name):
             with io.open(os.path.join(runtime_dir, file_name), encoding='utf-8') as f:
                 info = json.load(f)
 
-            # Simple check whether that process is really still running
+            # Actively check whether that process is really available via an HTTP request
             # Also remove leftover files from IPython 2.x without a pid field
-            if ('pid' in info) and check_pid(info['pid']):
+            response = kernel_request(info, path=url_path_join(info.get('base_url') or '/','/api'))
```

Comment:
Is the base-url stuff necessary?  It's not used here: https://github.com/jupyter/notebook/pull/5716/files#diff-bbd307804bb6b8836da9f06b8d8a06e9R498
```suggestion
            response = kernel_request(info, path='api')
```